projects/components/src/datagrid/datagrid.component.ts

```diff
@@ -711,6 +741,37 @@ export class DatagridComponent<R> implements OnInit, AfterViewInit {
         return this.paginationCallback(paginationData.firstItem + 1, paginationData.lastItem + 1, this.totalItems);
     }
 
+    /**
+     * Gives the render spec to create the detail row for the row with the given record, at the given index, and
+     * in a datagrid with the given count of total items.
+     */
+    getDetailRenderSpec(record: R, index: number, count: number): ComponentRendererSpec<DetailRowConfig<R>> {
```

Comment:
We can get this od this method

Reply:
Because the config object is still complicated, I moved this method into the new sub component.